docs/proposals/202005_scalable-rule-storage.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Scalable rule storage
+type: proposal
+menu: proposals
+status: proposed
+---
+
+## Summary
+
+There is no way to scale rule evaluation and storage today except functionally
+sharding rules onto multiple instances the `thanos rule` component. However, we
+have already solved scaling storage of time-series across multiple processes:
+`thanos receive`.
+
+To scale storing results of rule evaluations this proposal proposes to allow
+the `thanos rule` component to have a stateless mode, storing results of
+queries by sending it to a `thanos receive` hashring instead of storing them
+locally.
+
+## Motivation
+
+A few large rules can create a significant amount of resulting time-series,
+which should not be limited to what we can scale to with a single TSDB embedded
+in the rule component.
+
+## Goals
+
+Allow scaling storage of rule evaluations.
```

Comment:
what about evaluations itself? Not a second goal here? (:

Reply:
added :)